walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -20,15 +20,38 @@
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    ArrayList<TimeRange> conflicts = getConflictingEvents(events, request);
-    return getPossibleTimeRanges(conflicts, request.getDuration());
+    Collection<String> mandatoryAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    long duration = request.getDuration();
+
+    if (mandatoryAttendees.isEmpty()) {
+      ArrayList<TimeRange> optionalConflicts = getConflictingEvents(events, optionalAttendees);
+      return getPossibleTimeRanges(optionalConflicts, duration);
+    }
```

Comment:
Does this block do anything that the later lines don't? If mandatoryAttendees is empty, wouldn't line 33 add nothing to the list anyways?

Reply:
I think so, I'll try removing it and running the tests

Reply:
It does. I could probably add comments to justify my steps I guess

Reply:
Oh ok. Does it fail the `onlyOptionalAttendeesNoRoom` test? Maybe in line 39 we just have to add a check for `request.getAttendees().isEmpty()` too?

Reply:
Ohhhhhhh that makes a lot of sense!

Reply:
My main thought process was to avoid making calls that I didn't need to make but that definitely simplifies the code an makes it easier to read. great catch!

Reply:
It works!